### PR TITLE
feat: adds new commands and capabilities (sso, custom idp, etc.)

### PR DESCRIPTION
**Pull request by stockbal**
## Updated Login Flow

Due to the added options of SSO and Custom Origin the login flow is now the following:
1. Choose API Region
2. Select Login 
3. Enter temporary code
   - only relevant if *single-sign-on* is chosen at step 2)
4. Select Custom Origin  
    - Prompt is only shown if at least _one_ origin is maintained in config store
5. Select Org
6. Select Space

## Features

- new command for triggering a resetting of the target (org, space) via `btpcflogin t`
- new command for adding entries in the `passLogins` property of the config store via `btpcflogin add-login`
- new command for adding an origin of a custom identity provider to the cli config store via `btpcflogin add-origin`

## Structural Changes

- Commands are realized via npm module `commander`
- Regions were moved to `JSON` file at `/src/util/data/regions.json`
- `/src/index.ts` was refactored in smaller sub modules  
   - `/src/commands` - contains entry points for commands of cli
   - `/src/util` - contains helper-/sub modules for cli
   - `/src/index.ts` - now only entry point where cli is configured with `Commander`

## Miscellanous

- All files are formatted with prettier (added also to `package.json`).
- Build now contains only `.js` files

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -11,19 +11,23 @@
...
     "node": ">=14"
   },
-  "main": "src/index.js",
+  "main": "lib/index.js",
   "type": "module",
   "bin": {
-    "btpcflogin": "./src/index.js"
+    "btpcflogin": "./lib/index.js"
   },
   "scripts": {
```
Comment (MarcusSchoelzel) [inline package.json:19]: - Keep the prepare script, as this is automatically called during publish.
- Move the json file to a seperate folder that can be referenced by both folders (src and lib). This makes copying the files obsolete and the dependencies to copyfiles and rimraf aren't needed. And the clean srcipt is not needed.
- Add ts-node to the dev dependencies and start the app in local script from typescript files.

File: package.json [hunk 2]
```diff
@@ -35,8 +39,15 @@
     "@types/configstore": "^6.0.0",
     "@types/figlet": "^1.5.6",
     "@types/node": "^20.4.5",
+    "copyfiles": "^2.4.1",
+    "prettier": "^3.0.0",
+    "rimraf": "^5.0.1",
     "typescript": "^5.1.6"
   },
+  "prettier": {
+    "trailingComma": "none",
```
Comment (MarcusSchoelzel) [inline package.json]: Don't change the trailingComma setting

File: src/util/helper.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import util from "util";
+import child_process from "child_process";
+import { dirname } from "path";
+import { fileURLToPath } from "url";
+
+/**
+ * Promisified exec function
+ */
+export const exec = util.promisify(child_process.exec);
```
Comment (MarcusSchoelzel) [inline src/util/helper.ts]: Remove exec and replace it with spawn

**Comment by stockbal** [comment]
Requested changes have been applied.

**Note**:
Due to the changing of `exec` to `spawnSync`, [ora](https://www.npmjs.com/package/ora) was chosen a spinner library.

**Review by MarcusSchoelzel — CHANGES_REQUESTED** [review]
Please move "Origin" from config store to password manger. This makes the connection between the user and the idp clearer. And the step to choose the origin is not needed.

**Comment by stockbal** [comment]
Adjustments are made.

I also made some other refactorings as well:
- All cf cli calls are now located in a single class module
- singleton exports were removed
- error handler could be moved into cf cli module

**Review by MarcusSchoelzel — APPROVED** [review]

**Pull request merged**